tests/src/test/scala/akka/kafka/scaladsl/RebalanceExtSpec.scala

```diff
@@ -0,0 +1,427 @@
+  system.eventStream.subscribe(deadLetterListener, classOf[DeadLetter])
+
+  class RebalanceListenerActor(topicGroupId: String, logger: Logger) extends Actor with ActorLogging {
+    def receive: Receive = {
+      case TopicPartitionsAssigned(subscription, topicPartitions) =>
+        logger.info(
+          s"TopicPartitionsAssigned: self $self sender $sender topicGroupId $topicGroupId topicPartitions $topicPartitions subscription $subscription"
+        )
+      case TopicPartitionsRevoked(subscription, topicPartitions) =>
+        logger.info(
+          s"TopicPartitionsRevoked: self $self sender $sender topicGroupId $topicGroupId topicPartitions $topicPartitions subscription $subscription"
+        )
+    }
+  }
+
+  def assignmentHandler(clientId: String): PartitionAssignmentHandler = new PartitionAssignmentHandler {
+    override def onAssign(tps: Set[TopicPartition], consumer: RestrictedConsumer): Unit = {
+      log.debug(s"assignmentHandler::onAssign: client_id $clientId tps $tps consumer $consumer ")
+    }
+    override def onRevoke(tps: Set[TopicPartition], consumer: RestrictedConsumer): Unit = {
+      log.debug(s"assignmentHandler::onRevoke: client_id $clientId tps $tps consumer $consumer ")
+    }
+    override def onLost(tps: Set[TopicPartition], consumer: RestrictedConsumer): Unit = {
+      log.debug(s"assignmentHandler::onLost: client_id $clientId tps $tps consumer $consumer ")
+    }
+    override def onStop(tps: Set[TopicPartition], consumer: RestrictedConsumer): Unit = {
+      log.debug(s"assignmentHandler::onStop: client_id $clientId tps $tps consumer $consumer ")
+    }
+  }
+
+  def consumerSettings(group: String,
+                       maxPollRecords: String,
+                       partitionAssignmentStrategy: String): ConsumerSettings[String, String] =
+    consumerDefaults
+      .withGroupId(group)
+      .withCloseTimeout(5.seconds)
+      .withPollInterval(300.millis)
+      .withPollTimeout(200.millis)
+      .withProperty(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, maxPollRecords) // 500 is the default value
+      .withProperty(ConsumerConfig.PARTITION_ASSIGNMENT_STRATEGY_CONFIG, partitionAssignmentStrategy)
+
+  def subscribeAndConsumeMessages(
+      clientId: String,
+      perPartitionMessageCount: Int,
+      topicPartitionFutureMap: Map[String, Promise[Done]],
+      messageStoreAndAck: Map[Int, (AtomicInteger, Promise[Done], Promise[Done])],
+      consumerSettings: ConsumerSettings[String, String],
+      subscription: AutoSubscription,
+      tpCount: Int,
+      ptCount: Int,
+      sharedKillSwitch: SharedKillSwitch,
+      businessFlow: (String,
+                     Int,
+                     Map[String, Promise[Done]],
+                     Map[Int, (AtomicInteger, Promise[Done], Promise[Done])],
+                     SharedKillSwitch,
+                     String) => Flow[CommittableMessage[String, String], CommittableOffset, NotUsed]
```

Comment:
There's only 1 business flow (I assume you had more than 1 at some point). This parameter doesn't seem necessary.

Reply:
yes, have other test cases for my own dev needs, I can make it implicit if you strongly feel that I should :)

Reply:
done